torch/_inductor/ir.py

```diff
@@ -4680,53 +4705,105 @@ def require_stride1(cls, x):
+                        stride_order=get_stride_order(
+                            V.graph.sizevars.size_hints(x.get_layout().stride)
+                        )
+                        if is_stride_order_storage_and_layout(x, order)
+                        else order,
+                        allow_padding=allow_padding,
+                    )
+                    return x
+                else:
+                    # If the exact_strides is given, freeze the FlexibleLayout to a FixedLayout with the exact_strides.
+                    as_storage_and_layout(
+                        x,
+                        freeze=True,
+                        want_contiguous=False,
+                        stride_order=None,
+                        allow_padding=allow_padding,
+                        exact_strides=exact_strides,
+                    )
+                    return x
+            elif isinstance(x.get_layout(), FixedLayout) and (
+                order
+                and x.get_layout().is_stride_ordered(order)
+                or (
+                    exact_strides
+                    and (
+                        isinstance(x.data, ReinterpretView)
+                        or list(x.get_layout().stride) == exact_strides
                     )
-                    if is_stride_order_storage_and_layout(x, order)
-                    else order,
-                    allow_padding=allow_padding,
+                    and x.get_layout().is_stride_ordered(get_stride_order(V.graph.sizevars.size_hints(exact_strides)))
                 )
-                return x
-            elif isinstance(
-                x.get_layout(), FixedLayout
-            ) and x.get_layout().is_stride_ordered(order):
+            ):
                 return x
             elif isinstance(x.get_layout(), MutationLayoutSHOULDREMOVE):
                 if isinstance(x.get_layout().real_layout(), FlexibleLayout):
                     raise AssertionError(
                         "the MutationLayoutSHOULDREMOVE's real layout shouldn't be FlexibleLayout"
                     )
-                elif isinstance(
-                    x.get_layout().real_layout(), FixedLayout
-                ) and x.get_layout().real_layout().is_stride_ordered(order):
+                elif isinstance(x.get_layout().real_layout(), FixedLayout) and (
+                    (order and x.get_layout().real_layout().is_stride_ordered(order))
+                    or (
+                        exact_strides
+                        and list(x.get_layout().real_layout().stride)
+                        == list(exact_strides)
+                    )
+                ):
                     return x
 
         # TODO - Storage to InputBuffer
-        if isinstance(x, InputBuffer) and x.get_layout().is_stride_ordered(order):
+        # TODO - how about exact_strides?
+        if (
+            isinstance(x, InputBuffer)
+            and order
+            and x.get_layout().is_stride_ordered(order)
```

Comment:
looks suspicious

Reply:
added check for exact strides https://github.com/pytorch/pytorch/pull/130956/commits/14aee8f1beb64101d88b05a8f8b421d80c958da2